plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/DBPDataSourceFolder.java

```diff
@@ -36,4 +36,14 @@ public interface DBPDataSourceFolder extends DBPNamedObject2 {
     DBPDataSourceRegistry getDataSourceRegistry();
 
     boolean canMoveTo(DBPDataSourceFolder folder);
+
+    /**
+     * Specifies if the folder is not present on the file system, but should not be removed on data sources configuration reload
+     */
+    void setProtected(boolean value);
```

Comment:
I don't like that we have it in public interface.
We also can calculate this property (all folders which contain provided datasources are protected)

Reply:
I was surprised, but detecting datasources of the folder is much more obscure